python/tvm/relay/frontend/pytorch_neo.py

```diff
@@ -0,0 +1,1049 @@
+                    elif input_node.debugName() in self._consts.keys():
+                        list_shape.append(int(self._consts[input_node.debugName()]))
+                    else:
+                        pass
+                self._inputs_r[node_name] = _expr.var(node_name, shape=list_shape, dtype='float32')
+            elif node.kind() == "prim::GetAttr":
+                continue
+
+            self._add_op(node_name, node.kind(), node)
+
+    # Graph Helper Functions
+
+    def _add_op(self, op_name, operator, op_node):
+        """ Add an operator and its operators inputs to the graph and insert placeholders
+            where an input is a call node.
+
+        Parameters
+        ----------
+        op_name : string
+            The ID of the op node
+
+        operator : string
+            The kind of operator
+
+        op_node : PyTorch Node object
+            The full Node object for the op node
+
+        """
+        self._ops[(op_name, operator)] = op_node
+        input_list_r = []
+        for input_node in op_node.inputs():
+            if input_node.debugName() in self._inputs_r.keys():
+                input_list_r.append(self._inputs_r[input_node.debugName()])
+            elif input_node.debugName() in self._params.keys():
+                input_list_r.append(self._params[input_node.debugName()])
+            elif input_node.node().kind() == "prim::Constant":
+                input_list_r.append(self._consts[input_node.debugName()])
+            else:
+                input_list_r.append("call/var."+input_node.debugName())
+
+                # If the inputs of a ListConstruct op is a call or var, remove it from inputs
+                if op_node.kind() == 'prim::ListConstruct':
+                    if op_name in self._inputs_r.keys():
+                        self._inputs_r.pop(op_name)
+
+        self._op_inputs_r[(op_name, operator)] = input_list_r
+
+    def _parse_import_prerequisites(self):
+        """ Calculate the named preconditions from PyTorch graph.
+
+        Returns
+        -------
+        missing_operators : set object
+            Set of operator names which don't have their mapping in TVM,
+            i.e. which are not supported
+
+        """
+        missing_operators = set()
+        for node in self._trace.graph.nodes():
+            if node.kind() == "prim::Constant" or node.kind() == 'prim::ListConstruct' or \
+                    node.kind() == 'prim::GetAttr':
+                pass
+            else:
+                if any([node.kind() in _convert_map]):
+                    pass
+                else:
+                    missing_operators.add(node.kind())
+
+        return missing_operators
+
+def from_pytorch_neo(filename, input_shapes):
+    """ Load PyTorch model in the form of a trace object into relay.
+    The companion parameters will be handled automatically.
+
+    Parameters
+    ----------
+    trace : GraphDef object
```

Comment:
fix the parameters decription? (trace and shape here should be filename and input_shapes?)

Reply:
Done.